gmodule.lua

```diff
@@ -156,11 +155,15 @@ function gModule:share(gm, ...)
 end
 
 function gModule:training()
-	self:apply(function(module) module:training() end)
```

Comment:
why do we replace this with a loop everywhere. The new apply from nn.Module seems compatible, no?